test/system/test_api_interfaces.py

```diff
@@ -91,10 +91,62 @@ def test_delete_and_return_true(self):
             self.assertNotIn('Loopback0', config[0]['interfaces'])
 
     def test_delete_ethernet_raises_not_implemented_error(self):
-        with self.assertRaises(NotImplementedError):
-            for dut in self.duts:
+        for dut in self.duts:
+            with self.assertRaises(NotImplementedError):
                 dut.api('interfaces').delete(random_interface(dut))
 
+    def test_create_and_delete_ethernet_sub_interface(self):
```

Comment:
Just verified that EOS doesn't raise an error for this order of operations. The subinterface will not be functional until the primary port is in routed mode though.